File: src/de/hsh/dennis/model/AudioPlayer.java
package de.hsh.dennis.model;

import common.util.Logger;
import javazoom.jl.decoder.JavaLayerException;
import javazoom.jl.player.Player;

import java.io.*;

public class AudioPlayer {

    public static class MusicPlayer {

        private static Player player;
        static         Thread musikThreat;
        private static BufferedInputStream bis;
        private static boolean isPlaying = false;

        public static void play() {
            if ( musikThreat.isAlive() ) {musikThreat.stop();}
            if ( bis == null || musikThreat == null ) {
                Logger.log( "no file to play detected. Please load a .mp3 first." );
                return;
            }
            Logger.log( "music starts playing" );
            musikThreat.start();
            setIsPlaying(true);
        }

        public static void pause() {
            Logger.log( "music paused" );
            musikThreat.suspend();
            setIsPlaying(false);
        }

        public static void resume() {
            Logger.log( "music playing again" );
            setIsPlaying(true);
            musikThreat.resume();
        }

        public static void stop() {
            if ( musikThreat != null ) {
                musikThreat.stop();
                musikThreat = null;
            }
            if ( bis != null ) {bis = null;}
            setIsPlaying(false);
        }

        public static void shutdown() {
            stop();
            if ( player != null ) {

                player.close();
            }
            setIsPlaying(false);
            Logger.log( "MusikPlayer and dependencies finaly shut down !!!" );
        }

        public static void loadFile( String mp3Name ) {
            String path = "/de/hsh/dennis/resources/audioFiles/";

            InputStream fis =  AudioPlayer.class.getResourceAsStream(path + mp3Name);
            bis = new BufferedInputStream( fis );

            try {
                player = new Player( bis );
                musikThreat = new Thread( () -> {
                    try {
                        player.play();
                    }
                    catch ( JavaLayerException e ) {
                        e.printStackTrace();
                    }
                } );
            }
            catch ( JavaLayerException e ) {
                e.printStackTrace();
            }

        }


        public static boolean getIsPlaying() {
            return isPlaying;
        }

        public static void setIsPlaying(boolean playing) {
            isPlaying = playing;
        }
    }
}

File: Games/Dennis/src/de/hsh/dennis/model/AudioPlayer.kt
package de.hsh.dennis.model

import common.logger.Logger.log
import javazoom.jl.decoder.JavaLayerException
import javazoom.jl.player.Player
import java.io.BufferedInputStream

@Suppress("DEPRECATION")
class AudioPlayer {
    object MusicPlayer {
        private var player: Player? = null
        var musikThreat: Thread? = null
        private var bis: BufferedInputStream? = null
        var isPlaying = false
        fun play() {
            if (musikThreat!!.isAlive) {
                musikThreat!!.stop()
            }
            if (bis == null || musikThreat == null) {
                log("no file to play detected. Please load a .mp3 first.")
                return
            }
            log("music starts playing")
            musikThreat!!.start()
            isPlaying = true
        }

        fun pause() {
            log("music paused")
            musikThreat!!.suspend()
            isPlaying = false
        }

        fun resume() {
            log("music playing again")
            isPlaying = true
            musikThreat!!.resume()
        }

        fun stop() {
            if (musikThreat != null) {
                musikThreat!!.stop()
                musikThreat = null
            }
            if (bis != null) {
                bis = null
            }
            isPlaying = false
        }

        fun shutdown() {
            stop()
            if (player != null) {
                player!!.close()
            }
            isPlaying = false
            log("MusikPlayer and dependencies finaly shut down !!!")
        }

        fun loadFile(mp3Name: String) {
            val path = "/de/de.hsh/dennis/resources/audioFiles/"
            val fis = AudioPlayer::class.java.getResourceAsStream(path + mp3Name)
            bis = BufferedInputStream(fis)
            try {
                player = Player(bis)
                musikThreat = Thread(Runnable {
                    try {
                        player!!.play()
                    } catch (e: JavaLayerException) {
                        e.printStackTrace()
                    }
                })
            } catch (e: JavaLayerException) {
                e.printStackTrace()
            }
        }

    }
}
